helper.sh

```diff
@@ -81,6 +83,8 @@ get_type_direct() {
     fi
 }
 
+# like get_type, but use full componant json, instead of API function Name
```

Comment:
Retrieve the type directly from the full component JSON rather than using the API function Name in get_type.